sdk/servicebus/service-bus/review/service-bus.api.md

```diff
@@ -3,479 +3,563 @@
 
 // @public
-export type EntityStatus = "Active" | "Creating" | "Deleting" | "ReceiveDisabled" | "SendDisabled" | "Disabled" | "Renaming" | "Restoring" | "Unknown";
+export type EntityStatus =
+  | "Active"
+  | "Creating"
+  | "Deleting"
+  | "ReceiveDisabled"
+  | "SendDisabled"
+  | "Disabled"
+  | "Renaming"
+  | "Restoring"
+  | "Unknown";
+
+export { HttpOperationResponse };
 
-export { HttpOperationResponse }
+// @public (undocumented)
+export type Message = Omit<
+  ServiceBusMessage,
+  "complete" | "abandon" | "defer" | "deadletter" | "prototype"
+>;
 
 // @public
 export type MessageCountDetails = {
-    activeMessageCount: number;
-    deadLetterMessageCount: number;
-    scheduledMessageCount: number;
-    transferMessageCount: number;
-    transferDeadLetterMessageCount: number;
+  activeMessageCount: number;
+  deadLetterMessageCount: number;
+  scheduledMessageCount: number;
+  transferMessageCount: number;
+  transferDeadLetterMessageCount: number;
 };
 
 // @public
 export interface MessageHandlerOptions {
-    autoComplete?: boolean;
-    maxConcurrentCalls?: number;
-    maxMessageAutoRenewLockDurationInSeconds?: number;
+  autoComplete?: boolean;
+  maxConcurrentCalls?: number;
+  maxMessageAutoRenewLockDurationInSeconds?: number;
 }
 
-export { MessagingError }
+export { MessagingError };
+
+// @public
+export interface NonSessionReceiverTrack2<LockModeT extends "peekLock" | "receiveAndDelete"> {
+  // Warning: (ae-forgotten-export) The symbol "MessageIterator" needs to be exported by the entry point index.d.ts
+  //
+  // (undocumented)
+  iterateMessages(): MessageIterator<ContextType<LockModeT>>;
+  // Warning: (ae-forgotten-export) The symbol "MessageHandlers" needs to be exported by the entry point index.d.ts
+  // Warning: (ae-forgotten-export) The symbol "ContextType" needs to be exported by the entry point index.d.ts
+  //
+  // (undocumented)
+  streamMessages(handlers: MessageHandlers<ContextType<LockModeT>>): void;
+}
 
 // @public
 export interface OnError {
-    (error: MessagingError | Error): void;
+  (error: MessagingError | Error): void;
 }
 
 // @public
 export interface OnMessage {
-    (message: ServiceBusMessage): Promise<void>;
+  (message: ServiceBusMessage): Promise<void>;
 }
 
-// Warning: (ae-forgotten-export) The symbol "Client" needs to be exported by the entry point index.d.ts
-//
 // @public
-export class QueueClient implements Client {
-    close(): Promise<void>;
-    createReceiver(receiveMode: ReceiveMode): Receiver;
-    createReceiver(receiveMode: ReceiveMode, sessionOptions: SessionReceiverOptions): SessionReceiver;
-    createSender(): Sender;
-    readonly entityPath: string;
-    static getDeadLetterQueuePath(queueName: string): string;
-    readonly id: string;
-    peek(maxMessageCount?: number): Promise<ReceivedMessageInfo[]>;
-    peekBySequenceNumber(fromSequenceNumber: Long, maxMessageCount?: number): Promise<ReceivedMessageInfo[]>;
-}
+export type QueueAuth =
```

Comment:
Difference: we break out the individual connection string combinations into their various forms. Allows us to do better type inference in the constructors.

Reply:
This is gone now. The two-client model made it unnecessary. :)